File: Tutoriels/Python/POO/src/manager/GPrint.py
#================================================
class GPrint:
    #================================================
    __instance = None
    #================================================
    def __init__(self):
        if GPrint.__instance == None:
            GPrint.__instance = self        
    #================================================
    @staticmethod
    def Instance():
        if GPrint.__instance == None:
            GPrint()
        return GPrint.__instance
    #================================================
    def printData(self, data):
        print data      
#================================================

File: Tutoriels/CSharp/POO/src/manager/GPrint.cs
//===============================================
using System;
using System.Collections.Generic;
//===============================================
public class GPrint {
    //===============================================
    private static GPrint m_instance;
    static readonly object m_lock = new object();
    //===============================================
    private GPrint() {
		
    }
    //===============================================
    public static GPrint Instance() {
        lock(m_lock) {
            if(m_instance == null) {
               m_instance = new GPrint();
            }
	   	    return m_instance;
        }
    }    
    //===============================================
    public void print(string data) {
        Console.WriteLine(data);
    }
    //===============================================
}
//===============================================
